diff --git a/app/sap_80211_linux/Jamfile b/app/sap_80211_linux/Jamfile
index 0966abc..6e7e8f1 100644
--- a/app/sap_80211_linux/Jamfile
+++ b/app/sap_80211_linux/Jamfile
@@ -20,7 +20,6 @@ exe sap_80211
 	  timer_task.cpp
 	  linux/if_80211.cpp
 	  ../../lib/external/libnl//nlwrap
-	  ../../lib/external/libnl//libnl
 	  ../../lib/odtone//odtone
 	  /boost//program_options
 	: <link>static
diff --git a/lib/external/libnl/Jamfile b/lib/external/libnl/Jamfile
index 47585f5..6b3460f 100644
--- a/lib/external/libnl/Jamfile
+++ b/lib/external/libnl/Jamfile
@@ -15,25 +15,6 @@
 # This software is distributed without any warranty.
 #==============================================================================
 
-actions run_make
-{
-	make -C $(>:D) || exit
-
-	ar rcs $(<) `find $(>:D)/lib -type f -name \*.o`
-}
-
-make libnl.a
-	: libnl/configure
-	: @run_make
-	;
-
-lib libnl
-	:
-	: <file>libnl.a
-	:
-	: <include>libnl/include/
-	;
-
 lib nlwrap
 	: nlwrap/nl_cb.cpp
 	  nlwrap/nl_socket.cpp
@@ -43,7 +24,6 @@ lib nlwrap
 	  nlwrap/rtnl_socket.cpp
 	  nlwrap/rtnl_link.cpp
 	  nlwrap/rtnl_link_cache.cpp
-	  libnl
 	  ../../../lib/odtone//odtone
 	:
 	:
