diff --git a/app/sap_80211_linux/Jamfile b/app/sap_80211_linux/Jamfile
index 0966abc..6e7e8f1 100644
--- a/app/sap_80211_linux/Jamfile
+++ b/app/sap_80211_linux/Jamfile
@@ -20,7 +20,6 @@ exe sap_80211
 	  timer_task.cpp
 	  linux/if_80211.cpp
 	  ../../lib/external/libnl//nlwrap
-	  ../../lib/external/libnl//libnl
 	  ../../lib/odtone//odtone
 	  /boost//program_options
 	: <link>static
diff --git a/lib/external/libnl/Jamfile b/lib/external/libnl/Jamfile
index 47585f5..6b3460f 100644
--- a/lib/external/libnl/Jamfile
+++ b/lib/external/libnl/Jamfile
@@ -15,25 +15,6 @@
 # This software is distributed without any warranty.
 #==============================================================================
 
-actions run_make
-{
-	make -C $(>:D) || exit
-
-	ar rcs $(<) `find $(>:D)/lib -type f -name \*.o`
-}
-
-make libnl.a
-	: libnl/configure
-	: @run_make
-	;
-
-lib libnl
-	:
-	: <file>libnl.a
-	:
-	: <include>libnl/include/
-	;
-
 lib nlwrap
 	: nlwrap/nl_cb.cpp
 	  nlwrap/nl_socket.cpp
@@ -43,7 +24,6 @@ lib nlwrap
 	  nlwrap/rtnl_socket.cpp
 	  nlwrap/rtnl_link.cpp
 	  nlwrap/rtnl_link_cache.cpp
-	  libnl
 	  ../../../lib/odtone//odtone
 	:
 	:
diff --git a/lib/external/libnl/nlwrap/nl_socket.hpp b/lib/external/libnl/nlwrap/nl_socket.hpp
index 8e1ee9c..8ad4c3a 100644
--- a/lib/external/libnl/nlwrap/nl_socket.hpp
+++ b/lib/external/libnl/nlwrap/nl_socket.hpp
@@ -20,6 +20,7 @@
 
 #include <boost/noncopyable.hpp>
 
+#include <linux/types.h>
 #include <netlink/socket.h>
 #include "nl_msg.hpp"
 #include "nl_cb.hpp"
diff --git a/lib/external/libnl/nlwrap/rtnl_link_cache.cpp b/lib/external/libnl/nlwrap/rtnl_link_cache.cpp
index f84eed1..0963192 100644
--- a/lib/external/libnl/nlwrap/rtnl_link_cache.cpp
+++ b/lib/external/libnl/nlwrap/rtnl_link_cache.cpp
@@ -25,7 +25,7 @@ namespace nlwrap {
 
 rtnl_link_cache::rtnl_link_cache()
 {
-	if (::rtnl_link_alloc_cache(_socket, AF_UNSPEC, &_cache)) {
+	if (::rtnl_link_alloc_cache(_socket, &_cache)) {
 		throw std::runtime_error("Error allocating link cache");
 	}
 }
diff --git a/lib/external/libnl/nlwrap/rtnl_socket.cpp b/lib/external/libnl/nlwrap/rtnl_socket.cpp
index f5450d7..9bf0251 100644
--- a/lib/external/libnl/nlwrap/rtnl_socket.cpp
+++ b/lib/external/libnl/nlwrap/rtnl_socket.cpp
@@ -15,9 +15,11 @@
 // This software is distributed without any warranty.
 //==============================================================================
 
-#include "rtnl_socket.hpp"
-
+#include <linux/types.h>
 #include <netlink/route/link.h>
+#include <libnl/linux/netlink.h>
+
+#include "rtnl_socket.hpp"
 
 #include <stdexcept>
 
diff --git a/lib/external/libnl/nlwrap/rtnl_socket.hpp b/lib/external/libnl/nlwrap/rtnl_socket.hpp
index ad98ad0..fb817b4 100644
--- a/lib/external/libnl/nlwrap/rtnl_socket.hpp
+++ b/lib/external/libnl/nlwrap/rtnl_socket.hpp
@@ -20,7 +20,9 @@
 
 #include <boost/noncopyable.hpp>
 
+#include <linux/types.h>
 #include <netlink/socket.h>
+#include <libnl-tiny/linux/netlink.h>
 #include "nl_socket.hpp"
 
 namespace nlwrap {
